mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java

```diff
@@ -61,6 +62,9 @@
       }
     }
 
+    String[] readColumns = ColumnProjectionUtils.getReadColumnNames(job);
```

Comment:
I have seen this in Hive code as a comment in ColumnProjectionUtils.getReadColumnIDs:
```
      // NOTE: some code uses this list to correlate with column names, and yet these lists may
      //       contain duplicates, which this call will remove and the other won't. As far as I can
      //       tell, no code will actually use these two methods together; all is good if the code
      //       gets the ID list without relying on this method. Or maybe it just works by magic.
```
Not sure if this comment is still true or it is already fixed. If we might get duplicates I am not sure how Iceberg will behave in this case. Might worth to check. Maybe with some query with different UDFs around the same column?

Reply:
We stumbled across that same gem of a comment when we originally implemented this in Hiveberg. This seemed to suggest that this method *won't* return duplicates and thus should be safe to use. We haven't seen any problems so far but yes, if there is some way to trigger the potential duplicates via a Hive query we could add a HiveRunner test for this. Then again, maybe it just works by magic ;)